plugins/modules/lke_cluster.py

```diff
@@ -290,6 +360,43 @@ def _create_cluster(self) -> Optional[LKECluster]:
                 msg="failed to create lke cluster: {0}".format(exception)
             )
 
+    def _attempt_update_acl(self, cluster: LKECluster) -> None:
+        """
+        Handles the update logic for an LKE cluster's control plane ACL configuration.
+        """
+        control_plane_acl = self._safe_get_cluster_acl(cluster)
+        configured_acl = copy.deepcopy(self.module.params.get("acl"))
+
+        # We don't want to make any changes if the user has not explicitly defined an ACL
+        if configured_acl is None:
+            return
+
+        # [] and null are equivalent values for address fields,
+        # so we need to account for this when diffing
+        configured_addresses = configured_acl.get("addresses")
+        if configured_addresses is not None:
+            ipv4 = configured_addresses.get("ipv4")
+            ipv6 = configured_addresses.get("ipv6")
+
+            configured_acl["addresses"]["ipv4"] = None if ipv4 == [] else ipv4
+            configured_acl["addresses"]["ipv6"] = None if ipv6 == [] else ipv6
+
+        user_defined_keys = set(linode_lke_cluster_acl.keys())
+        current_acl = control_plane_acl if control_plane_acl is not None else {}
+
+        # Only diff on keys that can be defined by the user
+        current_acl = {
+            k: v for k, v in current_acl.items() if k in user_defined_keys
+        }
+
+        if configured_acl == current_acl:
+            return
```

Comment:
This diff logic is super messy but I can't think of a better way to implement it. Any feedback would be much appreciated!